server/src/main/java/io/crate/planner/operators/NestedLoopJoin.java

```diff
@@ -75,6 +75,7 @@ public class NestedLoopJoin implements LogicalPlan {
     private final Map<LogicalPlan, SelectSymbol> dependencies;
     private boolean orderByWasPushedDown = false;
     private boolean rewriteFilterOnOuterJoinToInnerJoinDone = false;
+    private boolean joinConditionOptimised = false;
```

Comment:
Could be final(?)

Is it too expensive to search for constant conditions as part of the rule pattern to see if the rule matches?
I'm not a fan of adding extra fields to detect whether a rule was already processed - but okay to keep if it's due to performance.